File: evals/refs/bs.py
import sys, math
def N(x):
    return 0.5 * (1 + math.erf(x / math.sqrt(2)))
data = sys.stdin.read().split()
T = int(data[0])
out = []
i = 1
for _ in range(T):
    S = float(data[i]); K = float(data[i+1]); T_y = float(data[i+2])
    r = float(data[i+3]); sigma = float(data[i+4])
    i += 5
    d1 = (math.log(S/K) + (r + 0.5*sigma*sigma)*T_y) / (sigma*math.sqrt(T_y))
    d2 = d1 - sigma*math.sqrt(T_y)
    c = S*N(d1) - K*math.exp(-r*T_y)*N(d2)
    out.append(f'{c:.4f}')
sys.stdout.write('\n'.join(out) + '\n')

File: evals/refs/bs.go
package main

import (
	"bufio"
	"fmt"
	"math"
	"os"
	"strconv"
)

func N(x float64) float64 {
	return 0.5 * (1 + math.Erf(x/math.Sqrt(2)))
}

func main() {
	in := bufio.NewScanner(os.Stdin)
	in.Buffer(make([]byte, 1<<20), 1<<20)
	in.Split(bufio.ScanWords)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()
	read := func() string { in.Scan(); return in.Text() }
	T, _ := strconv.Atoi(read())
	for c := 0; c < T; c++ {
		S, _ := strconv.ParseFloat(read(), 64)
		K, _ := strconv.ParseFloat(read(), 64)
		Ty, _ := strconv.ParseFloat(read(), 64)
		r, _ := strconv.ParseFloat(read(), 64)
		sigma, _ := strconv.ParseFloat(read(), 64)
		d1 := (math.Log(S/K) + (r+0.5*sigma*sigma)*Ty) / (sigma * math.Sqrt(Ty))
		d2 := d1 - sigma*math.Sqrt(Ty)
		price := S*N(d1) - K*math.Exp(-r*Ty)*N(d2)
		fmt.Fprintf(out, "%.4f\n", price)
	}
}
